Update Mongodb -> aggergation-sortbycount.md (#1844)
enkidevs/curriculum: mongodb/intro/aggregation-ii/aggregation-sortbycount-ii.md

---
author: Stefan-Stojanovic

levels:
  - beginner

aspects:
  - new
  - deep
  - workout

type: normal

category: tip

---

# Aggregating with $sortByCount and $match

---
## Content

> Previously we used only `$sortByCount`, now we can use the `$match` aggregation stage to only match documents of a certain `type` and perform the `$sortByCount: "$secondType"` for documents that match our specified `type`.

Reviewer: We should get out of the habit of using "Previously..." or "In the previous insight..." from now on (when writing new insights or when fixing old insights). We should try to make insights as individual as possible. This might not be possible in some cases, but there we can quickly sumarize the concept either in a phrase or a footnote.

In this case, we can just entirely skip the "Previously.." introduction:
```suggestion
We can use the `$match` aggregation stage to only match documents of a certain `type` followed by sorting by count for documents that match our specified `type`.
```

> For instance, let's say we want to count and sort documents by their `secondType`, but we are only interested in documents where the `type` is `"Water"`. We can do so like this:

Reviewer: how does this look?
```suggestion
Let's say we want to count and sort documents by their `secondType`, but we are only interested in documents where the `type` is `"Water"`. We can do so like this:

```
